### PR TITLE
Improve "Using callbacks" documentation section

### PR DESCRIPTION
**Pull request by taketwo**
I did not use skorch for several months, so I had to catch up with the new features and read documentation. This PR make some minor improvements and clarifications to the "Using callbacks" section.

### CODE REVIEW

**Comment by thomasjpfan** [comment]
Welcome back to skorch!

This PR LGTM 👍

File: docs/user/save_load.rst [hunk 1]
```diff
@@ -199,15 +198,17 @@ learning rate by using :class:`.LoadInitState`:
...
          17        0.1576       0.9453        0.1804     +  0.0184
 
-Since we started from the previous checkpoint which ended at epoch 7,
-the second run starts at epoch 8, continuing from the first
-checkpoint. With a lower learning rate, the validation loss was able
-to improve!
+The :class:`.LoadInitState` callback is executed once in the beginning
+of the training procedure and initializes model, history, and
+optimizer parameters from a specified checkpoint (if it exists). In
+our case previous checkpoint was created at the end of epoch 7, so the
```
Comment (BenjaminBossan) [inline docs/user/save_load.rst]: > our case previous checkpoint

our case, the previous checkpoint

**Comment by BenjaminBossan** [comment]
Apart from the minor comment, LGTM.

And: welcome back!

**Comment by taketwo** [comment]
I assume you would want to squash in the end anyway, so I pushed the fix as a separate commit to ease reviewing.

Thanks for the warm welcome everyone :)

**Comment by BenjaminBossan** [comment]
Yes, I personally prefer to have the individual commits during review and squash them later if need be.

LGTM, @ottonemo what do you think?

**Comment by BenjaminBossan** [comment]
Thank you.

**Pull request merged**